src/cpp/session/SessionSourceDatabase.cpp

```diff
@@ -71,13 +74,30 @@ namespace {
 // lookup)
 std::map<std::string, std::string> s_idToPath;
 
+// cached mapping of document last write times
+std::map<std::string, std::time_t> s_lastWriteTimes;
```

Comment:
Does this need to be a threadsafe map?

Reply:
No - all of these routines execute only on the main thread (thread 0)